### PR TITLE
DMP-3054: Fixing courtroom doesn't exist when receiving darts event

### PR DESCRIPTION
**Pull request by tonymort**
Fixing courtroom doesn't exist when receiving darts event

https://tools.hmcts.net/jira/browse/DMP-3054

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by jackmaloney — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/StandardEventHandlerTest.java [hunk 1]
```diff
@@ -269,6 +269,37 @@ void createsAnEventLinkedCaseForEachCaseNumberFromTheDartsEvent() {
             .containsOnly(idFrom(persistedEvents));
     }
 
+    @Test
+    void createsAnEventLinkedCaseForWhenCourtroomDoesntExist() {
```
Comment (davet1985) [inline src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/StandardEventHandlerTest.java:273]: Is this the intended name?
```suggestion
    void createsAnEventLinkedCaseWhenCourtroomDoesntExist() {
```

File: src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/StopAndCloseHandlerTest.java [hunk 2]
```diff
@@ -537,6 +537,77 @@ void shouldCreateNewCaseRetentionWithDefaultPolicyWhenNotDefined() {
...
+        assertThat(eventLinkedCases)
+            .extracting("courtCase.caseNumber")
+            .containsExactly(SOME_CASE_NUMBER, SOME_CASE_NUMBER_2);
+
+        assertThat(eventLinkedCases)
+            .extracting("event.id")
+            .containsOnly(idFrom(persistedEvents));
+    }
+
+    public Integer idFrom(List<EventEntity> eventEntities) {
```
Comment (davet1985) [inline src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/StopAndCloseHandlerTest.java:607]: private?

**Review by davet1985 — APPROVED** [review]
Couple of minor comments.

I don't actually get how this has fixed the issue? Looks like you have moved the courtroom creation from one place to another but it still happens before saving the event?

File: src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/StandardEventHandlerTest.java (hunk 1 above)
Comment (tonymort) [reply src/integrationTest/java/uk/gov/hmcts/darts/event/service/impl/StandardEventHandlerTest.java:273]: I'll take that

**Pull request merged**